### PR TITLE
Enable writing to OS CA trust

### PR DESCRIPTION
**Pull request by iw-gtadam**

### CODE REVIEW

File: src/acm_common/truststore_utils.py [hunk 1]
```diff
@@ -125,6 +125,35 @@ def generate_keystore(
...
+):
+    """Add certificate and private key.
+
+    Args:
+        priv_key (str): The base64 PEM-encoded private key to store
+        cert_list (List of str): A list of base64 PEM-encoded certificates
+            signed by ACM PCA.
+            Multiple certificates should represent the chain in the right order.
+        alias (str): The alias under which to store the key pair
+        priv_key_password (str): The password to protect the private key with
```
Comment (mattburgess) [inline src/acm_common/truststore_utils.py]: This parameter isn't used. It's rare (in my experience) to have private keys on disk protected by a passphrase, so I'd be happy to just see this removed from the function signature.

File: src/acm_common/truststore_utils.py [hunk 2]
```diff
@@ -125,6 +125,35 @@ def generate_keystore(
...
+
+    """
+    logger.info("Writing certificate and private key to filesystem")
+
+    f = open("/etc/pki/tls/private/" + alias + ".key", "a")
+    f.write(priv_key)
+    f.close()
+
+    for cert in cert_list:
+        f = open("/etc/pki/tls/certs/" + alias + ".crt", "a")
```
Comment (mattburgess) [inline src/acm_common/truststore_utils.py]: This is slightly more pythonic written as:

```
with open("/etc/pki/tls/private/" + alias + ".key", "a") as f:
   f.write(priv_key)
```
(note there's no need for an explicit close; it'll be closed automatically when the `with` block completes)

**Review by mattburgess — CHANGES_REQUESTED** [review]

File: src/acm_common/truststore_utils.py (hunk 1 above)
Comment (iw-gtadam) [reply src/acm_common/truststore_utils.py]: Updated to remove password from function

File: src/acm_common/truststore_utils.py (hunk 2 above)
Comment (iw-gtadam) [reply src/acm_common/truststore_utils.py]: Updated filesystem writes to use with block

**Review by mattburgess — APPROVED** [review]

**Pull request merged**